x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexer.java

```diff
@@ -461,4 +562,37 @@ private boolean checkState(IndexerState currentState) {
         }
     }
 
+    private synchronized void reQueueThrottledSearch() {
+        if (scheduledNextSearch != null && scheduledNextSearch.cancel()) {
```

Comment:
Similar comment to above, a current search could be inflight. This means that if the next delay is `0L`, we could have two searches occurring in parallel. 

As long as this method is NEVER called out of band, I think this might be ok.